src/wirecraft/client/ui/device.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pygame
+
+from .assets import SWITCH_DEVICE
+
+if TYPE_CHECKING:
+    from ..constants import RED
+    from .camera import Camera
+
+
+class Device(pygame.sprite.Sprite):
+    def __init__(self, world_pos: tuple[float, float], device_type: str):
+        super().__init__()
+        self.world_pos = world_pos
+        self.base_image = SWITCH_DEVICE
+        self.image = self.base_image
+        self.rect = self.image.get_rect()
+
+    def update_zoom(self, camera: Camera):
+        """Update device scale based on camera zoom"""
+        scaled_width = int(self.base_image.get_width() * camera.zoom)
+        scaled_height = int(self.base_image.get_height() * camera.zoom)
+        self.image = pygame.transform.scale(self.base_image, (scaled_width, scaled_height))
+        self.rect = self.image.get_rect()
+
+    def update_position(self, camera: Camera, screen_size: tuple[int, int]):
+        """Update device position based on camera"""
+        screen_pos = camera.world_to_screen(self.world_pos, screen_size)
+        self.rect.center = (int(screen_pos[0]), int(screen_pos[1]))
+
+    def draw(self, surface: pygame.Surface) -> None:
+        # draw the rect in red so we can see it
+        pygame.draw.rect(surface, RED, self.rect)
```

Comment:
Unsure whether this should be removed. This could help detect size issues such as with the switch but should not be shipped in final versions